install/quick-start/generate-thunder-secrets.sh

```
@@ -0,0 +1,99 @@
+# ---------------------------------------------------------------------------
+# Read existing secret if present — reuse values, only fill in missing keys
+# ---------------------------------------------------------------------------
+read_existing_secret() {
+    local key="$1" default="$2"
+    kubectl get secret openchoreo-initial-credentials -n thunder \
+        -o jsonpath="{.data.${key}}" 2>/dev/null | base64 -d 2>/dev/null || echo "$default"
+}
+
+SECRET_EXISTS=false
+if kubectl get secret openchoreo-initial-credentials -n thunder >/dev/null 2>&1; then
+    SECRET_EXISTS=true
```

Shellcheck (0.11.0)

[warning] SECRET_EXISTS appears unused. Verify use (or export if used externally).

(SC2034)